pandas/core/series.py

```diff
@@ -840,11 +840,15 @@ def _slice(self, slobj: slice, axis: int = 0) -> "Series":
 
     def __getitem__(self, key):
         key = com.apply_if_callable(key, self)
+        key = lib.item_from_zerodim(key)
 
         if key is Ellipsis:
             return self
 
-        key_is_scalar = is_scalar(key)
+        # check for is_list_like/slice instead of is_scalar to allow non-standard
+        #  scalars through, e.g. cftime.datetime needed by xarray
+        #  https://github.com/pydata/xarray/issues/3751
+        key_is_scalar = not is_list_like(key) and not isinstance(key, slice)
```

Comment:
this is really really sketch, why don't we just fix is_scalar?

also pls add a test if you can for this

Reply:
> why don't we just fix is_scalar?

Depends on what we want `is_scalar` to mean.  We could re-write `is_scalar` to just match this (actually more performant than what we have now), but it isn't really viable to add checks for any custom scalar that downstream libraries might implement.
 
Going to wait to hear from the downstream folks on if this actually solves their problem(s)

Reply:
I actually think is_scalar should just be this its much more generic and future proof

Reply:
hmm so changing is_scalar to just match this is breaking a bunch of tests bc `is_scalar(some_lambda_func)` is returning True

Reply:
right you might need some more exclusions, e.g. callabes

Reply:
so the trouble im facing ATM is that TimeGrouper is being recognized as a scalar

Reply:
The more I look at this the more skeptical I am of trying to amend is_scalar

Reply:
since we don't actually have a test case for this, am inclined to close.

Reply:
coming up with a test case isnt difficult, just need to settle on an approach.